File: website/models/WebhookLogEntry.ts
import { ObjectId } from 'mongodb';
import { z } from 'zod';

export type WebhookLogEntry = {
    _id: Date
    webhookId: ObjectId
    senderIp: string
    ownerId: ObjectId
    data: unknown
    successfulForwards: number
};

export const webhookLogEntrySchema = z.object({
    _id: z.optional(z.string()),
    webhookId: z.string(),
    senderIp: z.string(),
    data: z.any(),
    successfulForwards: z.number()
});

File: webhookServer/models/WebhookLogEntry.go
package models

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type WebhookLogEntry struct {
	Id                 time.Time              `bson:"_id,omitempty"`
	WebhookId          primitive.ObjectID     `bson:"webhookId"`
	OwnerId            primitive.ObjectID     `bson:"ownerId"`
	SenderIp           string                 `bson:"senderIp"`
	Data               map[string]interface{} `bson:"data"`
	SuccessfulForwards int                    `bson:"successfulForwards"`
}
